### PR TITLE
For #7376 - Update canClearSelection to check if text is empty

### PR DESCRIPTION
**Pull request by ekager**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt [hunk 1]
```diff
@@ -275,4 +276,20 @@ class GeckoEngineViewTest {
...
+
+    @Test
+    fun `canClearSelection should return false for null selection or null selection text`() {
+        val engineView = GeckoEngineView(context)
+        engineView.currentGeckoView = mock()
+        engineView.currentSelection = mock()
+
+        // null selection returns false
+        whenever(engineView.currentSelection?.selection).thenReturn(null)
+        Assert.assertFalse(engineView.canClearSelection())
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt:288]: nit: We usually `import org.junit.Assert.assertFalse` so this is just `assertFalse`.

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt [hunk 2]
```diff
@@ -275,4 +276,20 @@ class GeckoEngineViewTest {
...
+
+        // null text returns false
+        val selection: GeckoSession.SelectionActionDelegate.Selection = mock()
+        whenever(engineView.currentSelection?.selection).thenReturn(selection)
+        Assert.assertFalse(engineView.canClearSelection())
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt:293]: just `assertFalse` :)

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt [hunk 3]
```diff
@@ -275,4 +276,20 @@ class GeckoEngineViewTest {
...
+    fun `canClearSelection should return false for null selection or null selection text`() {
+        val engineView = GeckoEngineView(context)
+        engineView.currentGeckoView = mock()
+        engineView.currentSelection = mock()
+
+        // null selection returns false
+        whenever(engineView.currentSelection?.selection).thenReturn(null)
+        Assert.assertFalse(engineView.canClearSelection())
+
+        // null text returns false
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt:290]: We should also test for the "empty selection text" case. Unfortunately, Mockito can't mock `Selection.text`, but we we're keeping our own subtypes for mocking various GV classes. Here's how this would look in this case:
https://github.com/csadilek/android-components/commit/ffefb0d27052e916c00e25029f84704e00c8edca
